File: tests/client/postback-engine.test.ts
import { describe, it, expect } from 'vitest';
import { PostBackEngine } from '../../src/client/postback-engine';
import { HttpClient } from '../../src/client/http-client';

describe('PostBackEngine', () => {
  it('should create instance with HttpClient', () => {
    const client = new HttpClient();
    const engine = new PostBackEngine(client);
    expect(engine).toBeInstanceOf(PostBackEngine);
    expect(engine.hasState()).toBe(false);
  });

  it('should clear state', () => {
    const client = new HttpClient();
    const engine = new PostBackEngine(client);
    engine.clearState();
    expect(engine.hasState()).toBe(false);
  });
});

File: vulms_dart/test/auth/postback_engine_test.dart
import 'package:test/test.dart';
import 'package:vulms_dart/src/client/postback_engine.dart';
import 'package:vulms_dart/src/networking/vulms_http_client.dart';
import 'package:vulms_dart/src/utils/logger.dart';

void main() {
  group('PostBackEngine', () {
    late VulmsHttpClient httpClient;
    late VulmsLogger logger;
    late PostBackEngine engine;

    setUp(() {
      httpClient = VulmsHttpClient(logger: const NoopLogger());
      logger = const NoopLogger();
      engine = PostBackEngine(
        httpClient: httpClient,
        logger: logger,
      );
    });

    tearDown(() {
      httpClient.dispose();
    });

    group('hasState', () {
      test('returns false initially', () {
        expect(engine.hasState(), isFalse);
      });
    });

    group('clearState', () {
      test('clearState resets state', () {
        engine.clearState();
        expect(engine.hasState(), isFalse);
      });

      test('clearState then hasState returns false', () {
        engine.clearState();
        expect(engine.hasState(), isFalse);
      });

      test('clearState on fresh engine does not throw', () {
        expect(() => engine.clearState(), returnsNormally);
      });
    });
  });
}
